packages/matter.js/src/device/Endpoint.ts

```diff
@@ -21,28 +17,41 @@ import { FixedLabelCluster, UserLabelCluster } from "../cluster/LabelCluster.js"
 import { ClusterClientObj } from "../cluster/client/ClusterClient.js";
 import { ClusterServerObj, ClusterServerObjForCluster } from "../cluster/server/ClusterServer.js";
 import { InteractionClient } from "../protocol/interaction/InteractionClient.js";
-import { AllClustersMap } from "../cluster/index.js";
+import { AllClustersMap, BridgedDeviceBasicInformationCluster } from "../cluster/index.js";
+import { BasicInformationCluster } from "../cluster/BasicInformationCluster.js";
+
+export interface EndpointOptions {
+    endpointId?: number;
+    uniqueId?: string;
+}
 
 export class Endpoint {
     private readonly clusterServers = new Map<number, ClusterServerObj<Attributes, Commands, Events>>();
     private readonly clusterClients = new Map<number, ClusterClientObj<Attributes, Commands, Events>>();
     private readonly childEndpoints: Endpoint[] = [];
     id: number | undefined;
-    name: string;
-
-    private descriptorCluster;
-
+    uniqueId: string | undefined;
+    name = "";
+    private structureChangedCallback: () => void = () => {/** noop until officially set **/ };
+
+    private descriptorCluster: ClusterServerObjForCluster<typeof DescriptorCluster>;
+
+    /**
+     * Create a new Endpoint instance.
+     *
+     * @param deviceTypes One or multiple DeviceTypeDefinitions of the endpoint
+     * @param options Options for the endpoint
+     */
     constructor(
         protected deviceTypes: AtLeastOne<DeviceTypeDefinition>,
-        endpointId?: number
+        options: EndpointOptions = {}
```

Comment:
Could this be `endpoint: number | EndpointOptions`?  Then other changes converting `4` to `{ endpointId: 4}` would be unnecessary.  Might be clearer if we had an `type EndpointId = number` so it could be `endpoint: EndpointId | EndpointOptions`.

Reply:
The main point here was to change from "sigle parameters that alone are optional to a options structure, so it is more easy to extend later if needed. because else it would end in stzarnge calls like "(..., undefined, "uniqe-id") just because you do not set the endpoint, id. I think using an object is common in javascript and also our lib already

Reply:
Yeah makes sense, I was just thinking `Endpoint(deviceTypes, 1)` would be a shortcut for `Endpoint(deviceTypes, { endpointId: 1 })`

Reply:
In fact I hope that noone will manage endpoint Ids himelf :-)) so the uniqueEndpointStorageKey-string woud be more the "first parameter" but Ialso do not want to limit devs.
But in general a "string | EndpointOptions" is a meehhh typing :-)

Reply:
Cool, I was wondering if the key would be dominant.  I prefer "meh typing" over "meh usage" but we can revisit once API usage becomes more clear as it would be backward compatible